src/ui/details_panel.rs

```diff
@@ -1,19 +1,28 @@
 use ratatui::{
     crossterm::event::{KeyCode, KeyEvent, KeyModifiers},
     layout::Rect,
-    text::Text,
-    widgets::{Paragraph, Wrap},
+    text::{Line, Text},
+    widgets::{Block, BorderType, Padding, Paragraph, Wrap},
 };
 
+use crate::ui::scrollbar;
+
 /// Details panel used for the right side of each tab.
 /// This handles scrolling and wrapping.
 pub struct DetailsPanel {
-    pub scroll: u16,
+    scroll: u16,
     height: u16,
     lines: u16,
     wrap: bool,
 }
 
+/// Transient object holding render data
+pub struct DetailsPanelRenderContext<'a> {
+    panel: &'a mut DetailsPanel,
+    title: Option<Line<'a>>, // Note: ratatui::Title is deprecated
```

Comment:
Let's not leave this in code. Though it's probably a good idea to write `todo` with this in places where `Title` is used

Reply:
Good point. I will remove the comment regarding ratatui::Title